kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -26,59 +27,59 @@ const ContentNodeResource = coreApp.resources.ContentNodeResource;
 const FacilityUserResource = coreApp.resources.FacilityUserResource;
 const SummaryLogResource = coreApp.resources.ContentSummaryLogResource;
 
+/**
+ * Helper function for _showChannelList
+ * @param {object} channel - to get recentActivity for
+ * @param {string} classId -
+ * @returns {Promise} that resolves channel with lastActive value in object:
+ *   { 'channelId': dateOfLastActivity }
+*/
+function channelLastActivePromise(channel, classId) {
+  const summaryPayload = {
+    channel_id: channel.id,
+    collection_kind: ReportConstants.UserScopes.CLASSROOM,
+    collection_id: classId,
+  };
 
-function _showChannelList(store, classId) {
+  // workaround for conditionalPromise.then() misbehaving
+  return new Promise(
+    (resolve, reject) => {
+      const getSumm = ContentSummaryResource.getModel(channel.root_id, summaryPayload).fetch();
+      getSumm.then(
+        channelSummary => {
+          resolve({ [channel.id]: channelSummary.last_active });
+        },
+        error => reject(error)
+      );
+    }
+  );
+}
+
+function getAllChannelsLastActivePromise(channels, classId) {
+  const promises = channels.map((channel) => channelLastActivePromise(channel, classId));
+  return Promise.all(promises);
+}
+
+function _showChannelList(store, classId, showRecentOnly = false) {
```

Comment:
I would prefer to have a `SET_RECENT_ONLY` mutation that we set [after showing the channel list](https://github.com/learningequality/kolibri/pull/1406/files#diff-fea9e3e75d8fe0e1e96da5074cd65528R338). It feels more explicit to me, but I think @indirectlylit should weigh in.